include/mbedtls/md.h

```diff
@@ -33,6 +33,12 @@
 #include MBEDTLS_CONFIG_FILE
 #endif
 
+#ifndef ALLOW_PRIVATE_ACCESS
```

Comment:
This would need to be called `MBEDTLS_ALLOW_PRIVATE_ACCESS` to avoid name clashes with other headers that may be included in applications. We must be very strict about namespaces in public headers.

And that means we need to tweak the `check-names` machinery so that it doesn't complain about an undefined macro.

Reply:
renamed to MBEDTLS_ALLOW_PRIVATE_ACCESS